src/analyses/variable-sensitivity/abstract_enviroment.cpp

```diff
@@ -483,9 +483,9 @@ bool abstract_environmentt::merge(const abstract_environmentt &env)
     // For each element in the intersection of map and env.map merge
     // If the result of the merge is top, remove from the map
     bool modified=false;
-    for(const auto &entry : env.map)
+    for(const auto &entry : env.map.get_view())
```

Comment:
The current implementation of `merge()` unfortunately breaks all the sharing. This is due to the use of `[]` (both in the argument to `merge()` and later when `new_object` is assigned). For retrieving the item to pass to `merge()`, the const version of `find()` should be used. Later, the write in line 498 should only happen when `new_object` is different to the item currently stored in the map.

Ideally, we would use `get_delta_view()` here which only iterates over the elements with keys being either in the first map or in both maps, and with the items not being shared between them. This avoid iterating over elements that are shared (which we don't have to look at since we know that they are the same).

Also, there seems to be a bug in the implementation of `merge()`. Items that are only in the first map should be removed from it (i.e., they should be set to top).